### ISSUE TITLE
more gracious form of updating sendAs in composer

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#2140]
We should do a more gracious form of updating sendAs in composer, since now it will also notice footer changes, and it's too obstructing to always ask user to "reload the window".

So that when sendAs changes, detected by the following method:

```js
  static refreshAcctAliases = async (acctEmail: string): Promise<boolean> => {
    const fetchedSendAs = await Settings.fetchAcctAliasesFromGmail(acctEmail);
    const { sendAs: storedAliases, addresses: oldStoredAddresses } = (await Store.getAcct(acctEmail, ['sendAs', 'addresses']));
    await Store.setAcct(acctEmail, { sendAs: fetchedSendAs });
    if (!storedAliases) { // Aliases changed (it was previously undefined)
      if (oldStoredAddresses) { // Temporary solution
        return Object.keys(fetchedSendAs).sort().join(',') !== oldStoredAddresses.sort().join(',');
      }
      return true;
    }
    if (Settings.getDefaultEmailAlias(fetchedSendAs) !== Settings.getDefaultEmailAlias(storedAliases)) { // Changed (default email alias was changed)
      return true;
    }
    if (Object.keys(fetchedSendAs).sort().join(',') !== Object.keys(storedAliases).sort().join(',')) { // Changed (added/removed email alias)
      return true;
    }
    if (Object.keys(fetchedSendAs).filter(email => fetchedSendAs[email].footer).map(email => fetchedSendAs[email].footer).sort().join(',') !==
      Object.keys(storedAliases).filter(email => storedAliases[email].footer).map(email => storedAliases[email].footer).sort().join(',')) {
      return true;
    }
    return false; // Nothing changed
  }
```

It should transparently update sender options and also the footer information for use in the same secure window immediately.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/pull/2134#issuecomment-545817610_

**Comment by michael-volynets** [comment]
Do you mean to make refreshing aliases/footer synchronously and don't show a modal window that something has changed? Am I right?

**Comment by tomholub** [comment]
I'm thinking to still load them asynchronously. In the rare situations that the aliases or footer has changed, instead of asking user to reload frame:

1) re-render the send-from selector
2) replace `footer` with the new footer

That way user will not even notice.

Also, I just noticed, we are not pulling `sendAs` for replies, so when I tested replying on the new version before I used `new message`, it was not aware of the new footer. We should make it also async-pull for replies.
